### PR TITLE
feat: add common sign-in resolver for simple email-to-email matching

### PR DESCRIPTION
**Pull request by pjungermann**
- `providers.google.resolvers.emailMatchingUserEntityEmail()`
- `providers.microsoft.resolvers.emailMatchingUserEntityEmail()`
- `providers.okta.resolvers.emailMatchingUserEntityEmail()`

Signed-off-by: Patrick Jungermann <Patrick.Jungermann@gmail.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by pjungermann** [comment]
as requested at the previous PR, I just amended the changeset.

**Comment by pjungermann** [comment]
@jhaals due to the changeset modification likely something to be merged before the next pre-release.

Of course, I can change it if it does not work out.

File: plugins/auth-backend/src/providers/okta/provider.ts [hunk 1]
```diff
@@ -279,6 +282,10 @@ export const okta = createAuthProviderIntegration({
      * Looks up the user by matching their email local part to the entity name.
      */
     emailLocalPartMatchingUserEntityName: () => commonByEmailLocalPartResolver,
+    /**
+     * Looks up the user by matching their email to the entity email.
+     */
+    emailMatchingUserEntityEmail: () => commonByEmailResolver,
```
Comment (jhaals) [inline plugins/auth-backend/src/providers/okta/provider.ts:288]:
```suggestion
    emailMatchingUserEntityProfileEmail: () => commonByEmailResolver,
```

to make this extra clear

**Review by jhaals — COMMENTED** [review]
Thanks for reminding me @pjungermann. Small name tweak for these would be great

**Comment by jhaals** [comment]
Thanks Patrick, I did the last tweak in a separate branch to make this release

**Comment by pjungermann** [comment]
Would have had time now. Thanks for taking care already @jhaals

**Pull request merged**